src/content/docs/essentials/database.mdx

```diff
@@ -0,0 +1,205 @@
+abstract interface class SpendingsDataSource {
+    Future<List<TableSpendings>> getAllSpendings();
+    Future<void> insertSpending(TableSpendings spending);
+    Future<void> deleteSpending(TableSpendings spending);
+    Future<void> updateSpending(TableSpendings spending);
+}
+@DriftAccessor(tables: [Spendings])
+class SpendingsDao extends DatabaseAccessor<AppDatabase, Spendings>
+    with _$SpendingsDaoMixin implements SpendingsDataSource {
+        SpendingsDao(AppDatabase db) : super(db);
+...
+}
+```
+
+The repository will be used as a layer between the data from our table and the business logic of our application.
+
+```dart
+abstract class SpendingsRepository {
+    Future<List<Spendings>> getAllSpendings();
+    Future<void> insertSpending(Spendings spending);
+    Future<void> deleteSpending(Spendings spending);
+    Future<void> updateSpending(Spendings spending);
+}
+class SpendingsRepositoryImpl implements SpendingsRepository {
+    final SpendingsDataSource _spendingsDataSource;
+    SpendingsRepositoryImpl(this._spendingsDataSource);
+...
+}
+```
+
+### Writing data
+
+After we have created a table and data source for our expenses, we can write some data in it,
+
+```dart
+    await database.into(spendings).insert(
+      SpendingsCompanion.insert(
+        price: Value(3.9),
+        title: 'Coffee',
+        currency: Value('USD'),
+        date: DateTime.now(),
+        receiptImage: Value(Uint8List.fromList([1, 2, 3, 4])),
+      ),
+    );
+```
+
+delete data,
+
+```dart
+    await database.delete(
+        spendings,
+        where: (tbl) => tbl.id.equals(1),
+    ).go();
+```
+
+or update data.
+
+```dart
+    await database.update(
+        spendings,
+        where: (tbl) => tbl.id.equals(1),
+        set: SpendingsCompanion(
+            price: Value(2.9),
+            title: 'Coffee',
+            currency: Value('USD'),
+            date: DateTime.now(),
+            receiptImage: Value(Uint8List.fromList([1, 2, 3, 4])),
+        ),
+    ).go();
+```
+
+More detailed information about data writing operations, you can find  
+[here](https://drift.simonbinder.eu/docs/dart-api/writes/).
+
+### Reading data
+
+To read data, we need to create a query that will return the data we need.
+
+For example, we can retrieve all the data from the table
+
+```dart
+    final allSpendings = await database.select(spendings).get();
+```
+
+Or can filter the data,
+
+```dart
+    final query = await database.select(spendings)
+        ..where((tbl) => tbl.id.equals(1));
+```
+
+sort the data,
+
+```dart
+    final query = await database.select(spendings)
+        ..orderBy(
+            (tbl) => OrderingTerm(expression: tbl.price, mode: OrderingMode.desc),
+        );
+```
+
+or limit the data.
+
+```dart
+    final query = await database.select(spendings)
+        ..limit(10);
+```
+
+No one will stop us from combining everything together, but that's covered
+[here](https://drift.simonbinder.eu/docs/dart-api/select/).
```

Comment:
To my mind, there is no need to tell about all the features of Drift. Leave only inserts, updates and selects (but make select with various modificators such as limit and ordering)